geospaas_harvesting/harvesters.py

```diff
@@ -83,24 +83,68 @@ def harvest(self):
                 break
 
 
-class PODAACHarvester(Harvester):
-    """Harvester class for PODAAC data (NASA)"""
+class WebDirectoryHarvester(Harvester):
+    """
+    class for harvesting online data sources that rely on webpages (and most of the time on opendap)
+
+    Child classes should also assign values to the ingester and crawler class attributes.
+    These values should be the crawler and ingester classes to be used in the _create_crawlers()
+    and _create_ingester() methods.
+    """
+    ingester = None
+    crawler = None
+
+    def __init__(self, **config):
+        super().__init__(**config)
+        if 'excludes' in config:
+            if not isinstance(config['excludes'], list):
+                raise HarvesterConfigurationError(
+                    "'excludes' field must be fed with a python list of excluded names ")
+
     def _create_crawlers(self):
-        return [
-            crawlers.OpenDAPCrawler(url, time_range=(self.get_time_range()))
-            for url in self.config['urls']
-        ]
+        if self.crawler is None:
+            raise HarvesterConfigurationError(
+                "The class of crawler has not been specified properly")
+        try:
+            return [
+                self.crawler(url, time_range=(self.get_time_range()),
+                             excludes=self.config.get('excludes', None))
+                for url in self.config['urls']
+            ]
+        except (KeyError, TypeError, AttributeError,) as error:
```

Comment:
`KeyError` and `AttributeError` should be removed, just like in `_create_ingester`
```suggestion
        except TypeError as error:
```

Reply:
Done in the next commit.